plugins/modules/consul_role.py

```diff
@@ -0,0 +1,491 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2022, Håkon Lerring
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: consul_role
+short_description: Manipulate Consul roles
+version_added: 7.2.0
+description:
+ - Allows the addition, modification and deletion of roles in a consul
+   cluster via the agent. For more details on using and configuring ACLs,
+   see https://www.consul.io/docs/guides/acl.html.
+author:
+  - Håkon Lerring (@Hakon)
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+options:
+  name:
+    description:
+      - A name used to identify the role.
+    required: true
+    type: str
+  state:
+    description:
+      - whether the role should be present or absent.
+    required: false
+    choices: ['present', 'absent']
+    default: present
+    type: str
+  description:
+    description:
+      - Description of the role.
+    required: false
+    type: str
+    default: ''
+  policies:
+    type: list
+    elements: dict
+    description:
+      - List of policies to attach to the role.
+      - Each element must have a "name" or "id" (or both) to identify the policy. See consul_policy for more info.
```

Comment:
```suggestion
      - Each element must have a C(name) or C(id) (or both) to identify the policy. See consul_policy for more info.
```
Are other fields allowed? If not, it's better to describe exactly the format of the dictionary by using `suboptions:` (and `options=...` in the argument spec).

Does `consul_policy` refer to the community.general.consul_policy module?

Reply:
Thanks for the feedback, I'll update the documentation with `suboptions`.

Correct, it's the `community.general.consul_policy` module, it's missing an M() macro.